plugins/content/filter.php

```diff
@@ -0,0 +1,19 @@
+<?php
+class filter_content extends moodle_text_filter {
+    public function filter($text, array $options = array()) {
+        $matches = array();
+        $pattern = '/\[osx-raise id="([0-9]*)"\]/';
+        preg_match_all($pattern, $text, $matches);
+
+        $new_html = $text;
+        foreach ($matches[1] as $id) {
+            $c = new curl;
+            $object = $c->get("http://contentapi/contents/{$id}");
```

Comment:
If you want to avoid hard coding this here, you can use the little hack employed for the `eventsapi` where:

* [Pass an environment variable](https://github.com/openstax/raise-moodle-spikes/blob/0f9792039f9b406fec9a8fb402e71552652d5318/docker-compose.yml#L23) in the `docker-compose.yml`
* [Read the variable](https://github.com/openstax/raise-moodle-spikes/blob/0f9792039f9b406fec9a8fb402e71552652d5318/plugins/events/classes/event_processor.php#L72) in the code